Kia: update required packages

incognitojam opened the pull request:
- LKAS is standard:
  - Ceed 2019
  - Forte 2019-21
  - Optima 2019
- SCC is Standard:
  - Niro Plug-in Hybrid 2018-19
- LKAS is NOT standard:
  - Seltos 2021 (reverts previous change)
- Rename Niro Electric to Niro EV to match Kia branding
- The ACC package was called "Advanced Smart Cruise Control" before MY2019. There was no LKAS package, only LDWS. This applies to:
  - Optima 2017
  - Optima Hybrid 2017
  - Sorento 2018
- Delete Forte 2018 as it does not have a Smart Cruise Control package, not clear whether it is supported or not

**Checklist:**
- [x] Ceed
- [x] EV6
- [x] Forte
- [x] K5
- [x] Niro
- [x] Niro EV
- [x] Niro Plug-in Hybrid
- [x] Optima
- [x] Seltos
- [x] Sorento
- [x] Stinger
- [x] Telluride

Resume from stop etc. will be reviewed separately.

incognitojam committed:
Kia: LKAS is standard on Ceed 2019

https://www.downeys.co.uk/newmodels/Ceed_V3_24_09_2019.pdf

incognitojam committed:
Kia: delete Forte 2018

It doesn't appear that this vehicle has Adaptive/Smart Cruise Control in
any form, which was later added in 2019.

https://cdn.dealereprocess.org/cdn/brochures/kia/2018-forte.pdf
https://cdn.dealereprocess.org/cdn/brochures/kia/2019-forte.pdf

incognitojam committed:
Kia: LKAS is standard on Forte 2019+

https://cdn.dealereprocess.org/cdn/brochures/kia/2019-forte.pdf
https://cdn.dealereprocess.org/cdn/brochures/kia/2020-forte.pdf

incognitojam committed:
Kia: rename Niro Electric to EV

https://www.kia.com/us/en/niro

Co-authored-by: Shane Smiskol <shane@smiskol.com>

incognitojam committed:
Kia: LKAS is standard on Niro PHEV 2018+

https://cdn.dealereprocess.org/cdn/brochures/kia/2018-niro.pdf
https://cdn.dealereprocess.org/cdn/brochures/kia/2019-niro.pdf

incognitojam committed:
Kia: update required package on Optima 2017

The ACC package on the Optima 2017 is named "Advanced Smart Cruise
Control". It also doesn't have an LKAS package, only LDWS is available.

https://cdn.dealereprocess.org/cdn/brochures/kia/2017-optima.pdf

Co-authored-by: Shane Smiskol <shane@smiskol.com>

incognitojam committed:
Kia: LKAS is standard on Optima 2019

https://cdn.dealereprocess.org/cdn/brochures/kia/2019-optima.pdf

incognitojam committed:
Kia: revert package change to Seltos 2021

LKAS is NOT a standard package on the Seltos 2021

https://cdn.dealereprocess.org/cdn/brochures/kia/2021-seltos.pdf

incognitojam committed:
Kia: update required package on Sorento 2018

Similar to the Optima 2017, the ACC package on the Sorento 2018 is named
"Advanced Smart Cruise Control". It also doesn't have an LKAS package,
only LDWS.

SCC and LKAS were introduced in MY2019.

https://cdn.dealereprocess.org/cdn/brochures/kia/2018-sorento.pdf

Co-authored-by: Shane Smiskol <shane@smiskol.com>

incognitojam committed:
update docs

sshane commented on selfdrive/car/tests/test_docs.py:
```diff
@@ -34,7 +34,8 @@ def test_naming_conventions(self):
           self.assertNotIn("phev", tokens, "Use `Plug-in Hybrid`")
           self.assertNotIn("hev", tokens, "Use `Hybrid`")
-          self.assertNotIn("ev", tokens, "Use `Electric`")
+          if car.make != "Kia":
+            self.assertNotIn("ev", tokens, "Use `Electric`")
           if "plug-in hybrid" in car.model.lower():
             self.assertIn("Plug-in Hybrid", car.model, "Use correct capitalization")
```
```suggestion
          if "plug-in hybrid" in car.model.lower():
            self.assertIn("Plug-in Hybrid", car.model, "Use correct capitalization")
          if car.make != "Kia":
            self.assertNotIn("ev", tokens, "Use `Electric`")
```

sshane commented on docs/CARS.md:
```diff
@@ -82,22 +82,21 @@ A supported vehicle is one that just works when you install a comma three. All s
-|Kia|Optima 2019|Smart Cruise Control (SCC) & LKAS|Stock|0 mph|0 mph|[![star](assets/icon-star-full.svg)](##)|[![star](assets/icon-star-full.svg)](##)|Hyundai G|
-|Kia|Seltos 2021|Smart Cruise Control (SCC)|openpilot|0 mph|0 mph|[![star](assets/icon-star-full.svg)](##)|[![star](assets/icon-star-full.svg)](##)|Hyundai A|
-|Kia|Sorento 2018|Smart Cruise Control (SCC) & LKAS|Stock|0 mph|0 mph|[![star](assets/icon-star-full.svg)](##)|[![star](assets/icon-star-full.svg)](##)|Hyundai C|
+|Kia|Niro Plug-in Hybrid 2018-19|Smart Cruise Control (SCC)|openpilot|10 mph|32 mph|[![star](assets/icon-star-full.svg)](##)|[![star](assets/icon-star-full.svg)](##)|Hyundai C|
```
It looks like both SCC and LKAS are standard on the 2018 PHEV (no FE trim on the PHEV)

sshane commented on selfdrive/car/hyundai/values.py:
```diff
@@ -143,34 +143,31 @@ class HyundaiCarInfo(CarInfo):
+    HyundaiCarInfo("Kia Niro EV 2022", "All", video_link="https://www.youtube.com/watch?v=lT7zcG6ZpGo", harness=Harness.hyundai_h),
   ],
-  CAR.KIA_NIRO_PHEV: HyundaiCarInfo("Kia Niro Plug-in Hybrid 2018-19", min_enable_speed=10. * CV.MPH_TO_MS, harness=Harness.hyundai_c),
+  CAR.KIA_NIRO_PHEV: HyundaiCarInfo("Kia Niro Plug-in Hybrid 2018-19", "Smart Cruise Control (SCC)", min_enable_speed=10. * CV.MPH_TO_MS, harness=Harness.hyundai_c),
```
```suggestion
  CAR.KIA_NIRO_PHEV: HyundaiCarInfo("Kia Niro Plug-in Hybrid 2018-19", "All", min_enable_speed=10. * CV.MPH_TO_MS, harness=Harness.hyundai_c),
```

It looks like both SCC and LKAS are standard on the 2018 PHEV (no FE trim on the PHEV, LX is the lowest) https://cars.usnews.com/cars-trucks/kia/niro/2018

incognitojam replied:
ah, knew I missed one :upside_down_face:

incognitojam committed:
Kia: SCC is standard on Kia Niro PHEV 2018-19

https://cdn.dealereprocess.org/cdn/brochures/kia/2018-niro.pdf

Co-authored-by: Shane Smiskol <shane@smiskol.com>

sshane commented on selfdrive/car/hyundai/values.py:
```diff
@@ -143,34 +143,31 @@ class HyundaiCarInfo(CarInfo):
   CAR.KIA_OPTIMA: [
-    HyundaiCarInfo("Kia Optima 2017", min_steer_speed=32. * CV.MPH_TO_MS, harness=Harness.hyundai_b),
-    HyundaiCarInfo("Kia Optima 2019", harness=Harness.hyundai_g),
+    HyundaiCarInfo("Kia Optima 2017", "Advanced Smart Cruise Control", min_steer_speed=32. * CV.MPH_TO_MS, harness=Harness.hyundai_b),
```
Not sure but it was `SCC + LKAS/LDWS` in the original PR. This might've meant LDWS let you send LKAS commands to apply torque.
```suggestion
    HyundaiCarInfo("Kia Optima 2017", "Advanced Smart Cruise Control & LDWS", min_steer_speed=32. * CV.MPH_TO_MS, harness=Harness.hyundai_b),
```

incognitojam replied:
It might also be required to have the LKAS camera fitted?

sshane replied:
these exist on the same trim, so it's fine to just put scc

sshane commented on selfdrive/car/hyundai/values.py:
```diff
@@ -143,34 +143,31 @@ class HyundaiCarInfo(CarInfo):
+  CAR.KIA_SELTOS: HyundaiCarInfo("Kia Seltos 2021", harness=Harness.hyundai_a),
   CAR.KIA_SORENTO: [
-    HyundaiCarInfo("Kia Sorento 2018", video_link="https://www.youtube.com/watch?v=Fkh3s6WHJz8", harness=Harness.hyundai_c),
+    HyundaiCarInfo("Kia Sorento 2018", "Advanced Smart Cruise Control", video_link="https://www.youtube.com/watch?v=Fkh3s6WHJz8", harness=Harness.hyundai_c),
```
```suggestion
    HyundaiCarInfo("Kia Sorento 2018", "Advanced Smart Cruise Control & LDWS", video_link="https://www.youtube.com/watch?v=Fkh3s6WHJz8", harness=Harness.hyundai_c),
```

sshane replied:
nvm, it's the same trim

incognitojam commented:
I'm not sure I've checked the Optima Hybrid yet

incognitojam committed:
Kia: update required package on Optima Hybrid 2017

Similar to the Optima 2017, the ACC package on he Optima Hybrid 2017 is
named "Advanced Smart Cruise Control". It also doesn't have an LKAS
pacakge, only LDWS.

https://cdn.dealereprocess.org/cdn/brochures/kia/2017-optimahybrid.pdf
